src/socket.rs

```diff
@@ -787,7 +787,7 @@ impl UtpSocket {
         let self_t_micro: u32 = now_microseconds();
         let other_t_micro: u32 = original.timestamp_microseconds();
         resp.set_timestamp_microseconds(self_t_micro);
-        resp.set_timestamp_difference_microseconds(abs_diff(self_t_micro, other_t_micro));
+        resp.set_timestamp_difference_microseconds(self_t_micro.wrapping_sub(other_t_micro));
```

Comment:
Originally we had code like this (but failing): https://github.com/meqif/rust-utp/commit/7ed764e43fa561a96ec87862396c7cb3a758ed0d

I guess "preserving direction" is the alright as it is also what libutp does: https://github.com/bittorrent/libutp/blob/49692b8dcbc0b96d282423afeb5d9ffde87ab741/utp_internal.cpp#L1985

Useful discussion: https://github.com/meqif/rust-utp/issues/17